File: Component_DiceGame/DiceGame_csharp/ComponentDiceGame/view/Console.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Console.view
{
	abstract class Console
	{
		abstract public void DisplayInstructions();

		public void DisplayDiceValue(int a_faceValue)
		{
			System.Console.WriteLine("{0}", a_faceValue);
		}

		abstract public bool WantsToPlay();

		public char GetInput()
		{
			return System.Console.ReadKey().KeyChar;
		}

		abstract public void DisplayResult(bool a_isWinner);
	}
}

File: MVC_O_DiceGame/DiceGame_ruby/view/Console.rb
module View
  class Console


    def display_instructions
      raise NotImplementedError
    end

    def display_result 
      raise NotImplementedError
    end

    def wants_to_quit?
      raise NotImplementedError
    end

    def display_dice_value value
        puts value
    end

  end
end
